eth/gasestimator/gasestimator.go

```diff
@@ -80,6 +80,11 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 			}
 			available.Sub(available, call.Value)
 		}
+		if opts.Config.IsCancun(opts.Header.Number, opts.Header.Time) && len(call.BlobHashes) > 0 {
+			blobBalanceUsage := new(big.Int).SetInt64(int64(len(call.BlobHashes) * params.BlobTxBlobGasPerBlob))
+			blobBalanceUsage.Mul(blobBalanceUsage, call.BlobGasFeeCap)
+			available.Sub(available, blobBalanceUsage)
```

Comment:
I think we need another check for `balance > blobBalanceUsage` here, otherwise it could go negative.

Reply:
You're right! Which one of the following options would you use for the error?

- Using `core.ErrInsufficientFunds` as error, keeping `insufficient funds for gas * price + value` as description,
-- Using `core.ErrInsufficientFunds` as error, changing description to  `insufficient funds for gas * price + blobGas * blobGasPrice + value` as description,
- creating and using a new error, `ErrEip4844InsufficientFunds` with `insufficient funds for gas * price + blobGas * blobGasPrice + value` as description ?

I guess option 3?

Reply:
Added option 1/ for now, let me know.